obspy/io/csv/tests/test_csv.py

```diff
@@ -0,0 +1,215 @@
+    assert not iocsv._is_csv(fname)
+    events = read_events(fname, 'CSV', skipheader=1, names=names)
+    events2 = read_events(fname, 'CSV', skipheader=1, names=incomplete_names)
+    assert len(events) == 1
+    assert str(events[0].origins[0].time) == '2023-05-06T19:55:01.300000Z'
+    assert len(events[0].magnitudes) == 1
+    assert len(events2) == 1
+    assert str(events2[0].origins[0].time) == '2023-05-06T19:55:01.300000Z'
+    assert len(events2[0].magnitudes) == 0
+
+
+def test_csv_incomplete_catalog():
+    events = read_events()
+    del events[0].magnitudes[0].magnitude_type
+    events[1].magnitudes = []
+    events[1].preferred_magnitude_id = None
+    events[1].origins[0].depth = None
+    events[2].origins = []
+    events[2].preferred_origin_id = None
+    with NamedTemporaryFile(suffix='.csv') as ft:
+        with pytest.warns(Warning):
+            events.write(ft.name, 'CSV')
+        events2 = read_events(ft.name, 'CSV')
+    assert len(events2) == 2
+    assert events2[0].origins[0].time == events[0].origins[0].time
+    assert events2[1].origins[0].depth is None
+    assert events2[1].origins[0].time == events[1].origins[0].time
+
+    fname = get_example_file('incomplete.csv')
+    assert iocsv._is_csv(fname)
+    events = read_events(fname)
+    assert len(events) == 2
+    assert events[0].origins[0].depth is None
+    assert len(events[0].magnitudes) == 0
+    assert events[1].origins[0].depth is None
+    assert len(events[1].magnitudes) == 1
+    assert events[1].magnitudes[0].mag == 10.0
+
+
+def test_csv_custom_fmt():
+    events = read_events()
+    with NamedTemporaryFile(suffix='.csv') as ft:
+        fname = ft.name
+        events.write(fname, 'CSV', fields='{lat:.5f} {lon:.5f}')
+        assert not iocsv._is_csv(fname)
+        data = np.genfromtxt(fname, names=True, delimiter=',')
+        assert len(data) == 3
+        assert len(data[0]) == 2
+
+
+def test_csv_empty():
+    assert not iocsv._is_csv(b'')
+    empty_cat = iocsv._read_csv(b'')
+    assert len(empty_cat) == 0
+
+
+def test_csz(check_compression=False):
+    events = read_events('/path/to/example.pha')
+    tempdir = gettempdir()
+    fname = os.path.join(tempdir, 'obbspycsv_testfile.csz')
+    with NamedTemporaryFile(suffix='.csz') as ft:
+        fname = ft.name
+
+        def _test_write_read(events, **kw):
+            events.write(fname, 'CSZ', **kw)
+            assert iocsv._is_csz(fname)
+            events2 = read_events(fname, check_compression=check_compression)
+            assert len(events2) == len(events)
+            for ev1, ev2 in zip(events, events2):
+                assert len(ev2.origins[0].arrivals) == \
+                    len(ev1.origins[0].arrivals)
+                assert len(ev2.picks) == \
+                    len(ev1.picks)
+        _test_write_read(events)
+        _test_write_read(events, compression=False)
+        try:
+            import zlib  # noqa: F401
+        except ImportError:
+            pass
```

Comment:
consider `suppress` from context lib here to make it more concise and clear.

Reply:
I did not change anything here for now